app/renderer/lib/appium-client.js

```diff
@@ -309,10 +314,26 @@ export default class AppiumClient {
     return {contexts, contextsError, currentContext, currentContextError};
   }
 
-  async getSourceUpdate () {
+  /**
+   *
+   * @param {string} appMode Current app mode
+   * @returns {object} Current page source result or error message
+   */
+  async getSourceUpdate (appMode) {
+    let pageSource;
+    const {client: {capabilities: {platformName}}} = this.driver;
+    if (this.enableMobileSauceCmd === true && ['ios', 'tvos'].includes(_.toLower(platformName)) && appMode === APP_MODE.NATIVE) {
```

Comment:
basically, this switch will be 'mode' change via `sourceOptions` or something instead of the platformName